tests/EntryPoints/Specials/SpecialApproverCategoriesTest.php

```diff
@@ -2,73 +2,60 @@
-use RequestContext;
 use FauxRequest;
 use PermissionsError;
-use Wikimedia\Rdbms\Database;
+use ProfessionalWiki\PageApprovals\EntryPoints\Specials\SpecialApproverCategories;
+use SpecialPageTestBase;
+use User;
 
 /**
  * @group Database
  * @covers \ProfessionalWiki\PageApprovals\EntryPoints\Specials\SpecialApproverCategories
  */
-class SpecialApproverCategoriesTest extends MediaWikiIntegrationTestCase {
-
-	protected function setUp(): void {
-		parent::setUp();
-
-		$userFactory = $this->getServiceContainer()->getUserFactory();
-		$userGroupManager = $this->getServiceContainer()->getUserGroupManager();
+class SpecialApproverCategoriesTest extends SpecialPageTestBase {
 
-		$testUser1 = $userFactory->newFromName( 'TestUser1' );
-		$testUser1->addToDatabase();
-		$userGroupManager->addUserToGroup( $testUser1, 'approvers' );
+	protected function newSpecialPage(): SpecialApproverCategories {
+		return new SpecialApproverCategories();
+	}
 
-		$testUser2 = $userFactory->newFromName( 'TestUser2' );
-		$testUser2->addToDatabase();
-		$userGroupManager->addUserToGroup( $testUser2, 'approvers' );
+	public function testNonAdminCannotAccessPage() {
+		$this->expectException( PermissionsError::class );
+		$this->getPageOutput( user: $this->getTestUser()->getUser() );
 	}
 
-	private function createRequestContext( $user, $requestParams = [], $isPost = false ) {
-		$context = new RequestContext();
-		$context->setUser( $user );
-		$context->setRequest( new FauxRequest( $requestParams, $isPost ) );
-		return $context;
+	private function getPageOutput( User $user = null, array $request = [] ): string {
+		[ $output ] = $this->executeSpecialPage(
+			'',
+			$request === [] ? null : $this->newPostRequest( $request ),
+			'qqx',
+			$user ?? $this->getTestSysop()->getUser()
+		);
+		return $output;
 	}
 
-	private function executeSpecialPage( RequestContext $context ) {
-		$specialPage = new SpecialApproverCategories();
-		$specialPage->setContext( $context );
-		$specialPage->execute( null );
-		return $context->getOutput()->getHTML();
+	private function newPostRequest( array $parameters ): FauxRequest {
+		return new FauxRequest( $parameters, true );
 	}
 
-	public function testApproverPageForAdmin() {
-		$context = $this->createRequestContext( $this->getTestSysop()->getUser() );
-		$output = $this->executeSpecialPage( $context );
+	public function testAdminCanAccessPage(): void {
+		$output = $this->getPageOutput( user: $this->getTestSysop()->getUser() );
 		$this->assertStringContainsString( '<table', $output, 'Expected HTML output with table' );
 	}
 
-	public function testSpecialPageAccessForNonAdmin() {
-		$context = $this->createRequestContext( $this->getTestUser()->getUser() );
-		$this->expectException( PermissionsError::class );
-		$this->executeSpecialPage( $context );
-	}
+	public function testAddApproverAction(): void {
+		$username = self::getTestUser()->getUser()->getName();
```

Comment:
@waximabbax much simpler way to create a user, since MediaWiki already provides this test utility